From 397d096c17d7945c9553e3c0c50ad4741009c2c6 Mon Sep 17 00:00:00 2001
From: Alice Mikhaylenko <alicem@gnome.org>
Date: Mon, 20 Jan 2025 18:20:25 +0400
Subject: [PATCH 18/33] combo-row: Set width-chars=1 for the item labels

Otherwise it gets clipped with really narrow characters, like 1, l or i.

Fixes https://gitlab.gnome.org/GNOME/libadwaita/-/issues/720


(cherry picked from commit a26417fbb65719d58d8d608d9f7227dcedf36f68)
---
 src/adw-combo-row.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/adw-combo-row.c b/src/adw-combo-row.c
index b18ce110..2ab16023 100644
--- a/src/adw-combo-row.c
+++ b/src/adw-combo-row.c
@@ -291,6 +291,7 @@ setup_item (GtkSignalListItemFactory *factory,
   gtk_label_set_xalign (GTK_LABEL (label), 0.0);
   gtk_label_set_ellipsize (GTK_LABEL (label), PANGO_ELLIPSIZE_END);
   gtk_label_set_max_width_chars (GTK_LABEL (label), 20);
+  gtk_label_set_width_chars (GTK_LABEL (label), 1);
   gtk_widget_set_valign (label, GTK_ALIGN_CENTER);
   gtk_box_append (GTK_BOX (box), label);
 
-- 
2.47.1

